saleor/graphql/checkout/types.py

```diff
@@ -371,15 +378,42 @@ class Checkout(ModelObjectType[models.Checkout]):
         required=True,
         deprecation_reason=(f"{DEPRECATED_IN_3X_FIELD} Use `updatedAt` instead."),
     )
-    user = graphene.Field("saleor.graphql.account.types.User")
-    channel = graphene.Field(Channel, required=True)
-    billing_address = graphene.Field("saleor.graphql.account.types.Address")
-    shipping_address = graphene.Field("saleor.graphql.account.types.Address")
-    note = graphene.String(required=True)
-    discount = graphene.Field(Money)
-    discount_name = graphene.String()
-    translated_discount_name = graphene.String()
-    voucher_code = graphene.String()
+    user = graphene.Field(
+        "saleor.graphql.account.types.User",
+        description="The user assigned to the checkout.",
+    )
+    channel = graphene.Field(
+        Channel,
+        required=True,
+        description="The channel for which checkout was created.",
+    )
+    billing_address = graphene.Field(
+        "saleor.graphql.account.types.Address",
+        description="The billing address of the checkout.",
+    )
+    shipping_address = graphene.Field(
+        "saleor.graphql.account.types.Address",
+        description="The shipping address of the checkout.",
+    )
+    note = graphene.String(required=True, description="The note for the checkout.")
+    discount = graphene.Field(
+        Money,
+        description=(
+            "The total discount is applied to the checkout. "
+            "Note: Only discount created via voucher are included in this field."
+        ),
+    )
+    discount_name = graphene.String(
+        description="The name of voucher assigned to the checkout."
+    )
+    translated_discount_name = graphene.String(
+        description=(
+            "The name of voucher assigned to the checkout in the customer's language."
```

Comment:
I would say that this will be only set when `Checkout.languageCode` is provided.

"Translation of the `discountName` field in the language set in `Checkout.languageCode` field. Note: this field is set automatically when `Checkout.languageCode` is defined; otherwise it's `null`" (null or empty string, I'm not sure now)